app/src/services/cpk/fns.ts

```diff
@@ -345,15 +584,117 @@ export const prepareCondition = async (params: PrepareConditionParams) => {
   return { ...params, conditionId }
 }
 
+/**
+ * Resolve condition
+ */
+
+interface ResolveConditionParams {
+  isConditionResolved: boolean
+  isScalar: boolean
+  numOutcomes: number
+  oracle: OracleService
+  question: Question
+  realitio: RealitioService
+  scalarLow: Maybe<BigNumber>
+  scalarHigh: Maybe<BigNumber>
+  transactions: Transaction[]
+}
+
+export const resolveCondition = async (params: ResolveConditionParams) => {
+  const {
+    isConditionResolved,
+    isScalar,
+    numOutcomes,
+    oracle,
+    question,
+    realitio,
+    scalarHigh,
+    scalarLow,
+    transactions,
+  } = params
+  if (!isConditionResolved) {
+    if (isScalar && scalarLow && scalarHigh) {
+      transactions.push({
+        to: realitio.scalarContract.address,
+        data: RealitioService.encodeResolveCondition(question.id, question.raw, scalarLow, scalarHigh),
+      })
+    } else {
+      transactions.push({
+        to: oracle.contract.address,
+        data: OracleService.encodeResolveCondition(question.id, question.templateId, question.raw, numOutcomes),
+      })
+    }
+  }
+
+  return params
+}
+
+/**
+ * Claim winnings
+ */
+
+interface ClaimWinningsParams {
+  isConditionResolved: boolean
+  question: Question
+  realitio: RealitioService
+  transactions: Transaction[]
+}
+
+export const claimWinnings = async (params: ClaimWinningsParams) => {
+  const { isConditionResolved, question, realitio, transactions } = params
+
+  if (!isConditionResolved) {
```

Comment:
Shouldn't this be `if (isConditionResolved)`?

Reply:
`resolveCondition` is the first function in the pipeline so by the time `claimWinnings` is called the condition is resolved + it is super weird but `claimWinnings` is only called once and it claims winnings for all bonders into their realitio balance, `withdrawRealitioBalance` withdraws individual winnings